src/reducers/sources.js

```diff
@@ -276,10 +276,24 @@ function updateSource(state: SourcesState, source: Object) {
 
   state.sources[source.id] = updatedSource;
 
+  updateExistingRelativeSource(state, source);
   updateSourceUrl(state, source);
   updateOriginalSources(state, source);
 }
 
+function updateExistingRelativeSource(state: SourcesState, source: Object) {
+  const relativeSources = { ...state.relativeSources };
+
+  for (const thread in relativeSources) {
+    relativeSources[thread] = { ...relativeSources[thread] };
+    if (relativeSources[thread][source.id]) {
+      const existingRelativeSource = relativeSources[thread][source.id];
+      const updatedRelativeSource = { ...existingRelativeSource, ...source };
+      state.relativeSources[thread][source.id] = updatedRelativeSource;
+    }
+  }
+}
+
```

Comment:
this change looks good. can we write some unit tests to demonstrate the bug and fix.

Reply:
Hey @jasonLaster, sorry for the late response. Was doing some reading on jest tests and forgot to comment. 

Do you still need help with the unit tests?

Reply:
I'd love if you could put a few unit tests together!

This is really impressive work @derek-li !